### DISCUSSION TITLE
Port becomes unavailable

### DISCUSSION THREAD

**Discussion by vascolp** [pybricks/support#1144] [Q&A]
Scenario:
1) A TehcnicHub with nothing connected
2) flash 3.3.0b7 (with pybricksdev)
3) Connect a Motor to port A
4) run the program below with pybricksdev. The program gets registed in the hub. The motor runs.
5) Run the program by pressing the hub button. The motor runs.
6) Do step 5 as many times as you want. Motor always runs (as expected).

Now, if you uncomment the shutdown() line, and do step 4 again, it will run. But then, any other run, either 4 or 5, will go to the exception: `OSError: [Errno 19] ENODEV: A sensor or motor is not connected to the specified port:`
Step 5 will always do the exception (red light).
You can run the program in another port but port A becomes unavailable ... until you flash firmware again.

This thing gets more complicated. I am having problems with pybricksdev BLE communications and it seems that, when pybricksdev gets stuck trying to run a program, the last port I have used in the hub becomes unavailable? I can´t explain this very welL... does it makes sense?

I think there is something wrong with shutdown() or maybe device cleanup?

```
from pybricks.pupdevices import DCMotor, Motor, Remote
from pybricks.parameters import Port, Direction, Stop, Button, Color
from pybricks.tools import wait
from pybricks.hubs import ThisHub
from pybricks import version

print(version)
hub=ThisHub()

try:
    hub.light.blink(Color.GREEN, (5, 300, 1000, 5))
    wait(1000)
    d=Motor(Port.B)
    d.run(100)
    wait(1000)
    hub.light.blink(Color.CYAN, (5, 300, 1000, 5))
    wait(1000)
    #d.close()
    wait(1000)
    #hub.system.shutdown()
except Exception as e:
    hub.light.blink(Color.RED, (5, 300, 1000, 5))
    wait(1000)
    raise



```

**Comment by laurensvalk** [comment]
Thank you. This has already been fixed. You can grab the latest firmware from here https://nightly.link/pybricks/pybricks-micropython/workflows/build/master or wait for the next beta to come out.

**Reply by vascolp** [reply]
Checked, it works!! Thanks!
